### PR TITLE
Fix HTML Entities regex

### PR DESCRIPTION
**Pull request by JonathanWolfe**
Caused jonathanwolfe/language-polymer#1. Just need a simple lookahead assertion.

### CODE REVIEW

**Comment by esdoppio** [comment]
Could you also update the specs?

File: grammars/html.cson [hunk 1]
```diff
@@ -363,7 +363,7 @@
   'entities':
     'patterns': [
       {
-        'begin': '(&)([a-zA-Z0-9]+|#[0-9]+|#x[0-9a-fA-F]+)'
+        'begin': '(&)([a-zA-Z0-9]+|#[0-9]+|#x[0-9a-fA-F]+)(?=;)'
```
Comment (JonathanWolfe) [inline grammars/html.cson]: I think that's the better way forward since entities aren't able to wrap around anything.

**Comment by JonathanWolfe** [comment]
What do you want updated/added to the spec?

**Comment by JonathanWolfe** [comment]
Swapped to use match instead of the begin/end semantic. Also corrected spec changes.

**Comment by DavidePastore** [comment]
@50Wliu thanks for the cc. The fact is that `&a` has a particular meaning in my implementation: you're in the middle of an entity and you could use this information to trigger autocomplete. Maybe we could try to preserve the first pattern and edit the second.

**Comment by JonathanWolfe** [comment]
@DavidePastore - I'm not a regex guru but I can't think of any way to have that functionality and not break the rest of the spec. It will constantly try to eat the rest of the document (or line at a minimum with a change) thinking you're still writing an entity. It is possible to preserve autocomplete but it would require you to write `&;` then go back and fill in the entity.

**Comment by 50Wliu** [comment]
I'm reverting this for now and will investigate alternative options later.

**Comment by DavidePastore** [comment]
@JonathanWolfe - Yes, you're right. I'm not a regex guru either.
@50Wliu You did the right thing 👍

**Pull request merged**